### PR TITLE
store uncompressed index size on cloudfiles storage extension

### PR DESCRIPTION
**Pull request by duilio**
Fix bug discussed here: https://groups.google.com/d/msg/druid-user/Bt2wiMn282Q/n4VYzvV4DQAJ

### CODE REVIEW

**Comment by drcrallen** [comment]
Thanks @duilio can you please modify the PR title to mention CloudFormation by full name ?

Can you also add a unit test which checks for this error?

**Comment by fjy** [comment]
:+1: I'm not sure how difficult it is to add the UT

do you  mind signing the CLA here? http://druid.io/community/cla.html

**Comment by duilio** [comment]
Hi @drcrallen, I guess the UT for this issue should be a generic one to check the metadata consistency of the segment stored by the storages. Cannot find such a thing in the source for the other storage extensions but I'm quite new to the codebase.

Is there any reason why a storage extension could actually set an index size different from the original uncompressed size?

**Comment by duilio** [comment]
@fjy just signed. 
@drcrallen PR renamed ;-)

**Comment by fjy** [comment]
@drcrallen I think if we want folks to write UT on this, we need to create UT first for the existing use cases

**Comment by drcrallen** [comment]
Ideally there would be some general UT way of doing it, but for the purposes of this PR I think a mock of the CF API and calling push directly is sufficient.

One key problem with a general UT for pushing is that the pushers do not have a unified way of how push is handled. As such there would need to be a lot of plugability layers to make sure that all deep storage paradigms work. It is not immediately obvious such an approach would be better than just having each deep storage implementation handle its own Push unit test.

File: extensions/cloudfiles-extensions/src/main/java/io/druid/storage/cloudfiles/CloudFilesDataSegmentPusher.java [hunk 1]
```diff
@@ -68,8 +68,9 @@ public String getPathForHadoop(final String dataSource)
   public DataSegment push(final File indexFilesDir, final DataSegment inSegment) throws IOException
   {
     final String segmentPath = CloudFilesUtils.buildCloudFilesPath(this.config.getBasePath(), inSegment);
-    final File zipOutFile = File.createTempFile("druid", "index.zip");
     final File descriptorFile = File.createTempFile("descriptor", ".json");
+    final File zipOutFile = File.createTempFile("druid", "index.zip");
+    final long indexSize = CompressionUtils.zip(indexFilesDir, zipOutFile);
```
Comment (drcrallen) [inline extensions/cloudfiles-extensions/src/main/java/io/druid/storage/cloudfiles/CloudFilesDataSegmentPusher.java]: This should be inside the try to make sure any IOException still properly cleans up zipOutFile

**Comment by drcrallen** [comment]
This will introduce a (minor) regression related to https://github.com/druid-io/druid/pull/2191#discussion-diff-48909271.

@fjy has agreed to include UTs for this PR and the fix for the regression in https://github.com/druid-io/druid/pull/2206

**Pull request merged**